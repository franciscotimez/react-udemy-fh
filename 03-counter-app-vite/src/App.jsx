import PropTypes from "prop-types";

const newMessage = 'Francisco';
const obj = { 
  title: 'Hola',
  message: 'Mundo cruel'
}

export const App = ({title, subTitle}) => {

  return (
    <>
      <h1>{ title }</h1>
      <code>{ JSON.stringify(obj) }</code>
      <p>{ subTitle }</p>
    </>
  );
};

App.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.number
}

App.defaultProps = {
  title: 'No Hay titulo',
  subTitle: 32
}