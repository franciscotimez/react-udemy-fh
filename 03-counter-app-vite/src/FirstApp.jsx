import PropTypes from "prop-types";

export const FirstApp = ({title, subTitle, name}) => {
  return (
    <>
    <h1 data-testid="test-title">{title}</h1>
    <p>{subTitle}</p>
    <p>{name}</p>
    </>
  )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
}

FirstApp.defaultProps = {
    name: 'Francisco Timez',
    subTitle: 'No hay subTitle'
}