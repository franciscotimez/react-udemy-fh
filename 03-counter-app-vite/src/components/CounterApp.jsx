import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ({ initValue }) => {

  const [value, setValue] = useState(initValue);

  const handleAdd = () => {
    // setValue(value + 1);
    setValue( (v) => v + 1 );
  };

  const handleSub = () => {
    setValue( (v) => v - 1 );
  };

  const handleReset = () => {
    setValue( initValue );
  };


  return (
    <>
      <h1>CounterApp</h1>
      <h2> {value} </h2>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleSub}>-1</button>
      <button onClick={handleReset}>Reset</button>
    </>
  );
};


CounterApp.propTypes = {
  initValue: PropTypes.number.isRequired
};