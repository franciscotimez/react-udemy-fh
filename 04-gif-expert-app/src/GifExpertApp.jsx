

export const GifExpertApp = () => {
  return (
    <div>GifExpertApp</div>
  )
}
