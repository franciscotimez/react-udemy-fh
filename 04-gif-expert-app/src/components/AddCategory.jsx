import { useState } from "react";

export const AddCategory = ({onAddCategory}) => {
  const [inputValue, setInputValue] = useState('');

  const handleChange = (event) => {
    setInputValue(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    const newCategory = inputValue.trim()
    
    if(newCategory.length > 1) {
      onAddCategory(newCategory)
    }
    setInputValue('')
  }

  return (
    <form onSubmit={handleSubmit}>
      <input
        type='text'
        placeholder='Buscar Gifs'
        onChange={handleChange}
        value={inputValue}
      />
    </form>
  );
};
