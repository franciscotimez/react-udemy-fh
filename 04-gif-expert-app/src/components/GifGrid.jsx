

export const GifGrid = ({category}) => {
  return (
    <div>{category}</div>
  )
}
