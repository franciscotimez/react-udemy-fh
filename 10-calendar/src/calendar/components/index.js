export * from './CalendarEvent'
export * from './NavBar'
export * from './CalendarModal'
export * from './FabAddNew'
export * from './FabDelete'