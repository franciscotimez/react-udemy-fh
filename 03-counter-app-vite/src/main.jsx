import React from 'react';
import ReactDOM from 'react-dom/client';
import { App } from './App';
import { CounterApp } from './components/CounterApp';

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp initValue={0} />
    </React.StrictMode>
);