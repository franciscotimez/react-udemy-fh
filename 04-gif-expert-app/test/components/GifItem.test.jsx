import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";


describe('Test of <GifItem />', () => {

    const image = {
        id: '1231',
        title: 'Titulo de la imagen',
        url: 'http://imagen/'
    };

    test('should be to match with snapshot', () => {
        const { container } = render(
            <GifItem
                {...image}
            />
        );
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar la imagen con el src y al alt', () => {
        render(
            <GifItem
                {...image}
            />
        );

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(image.url);
        expect(alt).toBe(image.title);

    });

    test('debe de mostrar el titulo en el componente', () => {
        render(
            <GifItem
                {...image}
            />
        );
        
        expect(screen.getByText(image.title)).toBeTruthy()

     })

});
