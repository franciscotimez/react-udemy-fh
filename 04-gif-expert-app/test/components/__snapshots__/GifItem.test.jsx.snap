// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <GifItem /> should be to match with snapshot 1`] = `
<div>
  <div
    class="car"
  >
    <img
      alt="Titulo de la imagen"
      src="http://imagen/"
    />
    <p>
      Titulo de la imagen
    </p>
  </div>
</div>
`;
